internal/clients/upcloud.go

```diff
@@ -67,6 +74,41 @@ func TerraformSetupBuilder(version, providerSource, providerVersion string) terr
 			"username": creds["username"],
 			"password": creds["password"],
 		}
+
+		// This configures the SDKv2 provider so that it knows how to get credentials and such
+		diag := provider.TerraformProvider.Configure(context.TODO(), &terraformsdk.ResourceConfig{
+			Config: ps.Configuration,
+		})
+
+		if diag != nil && diag.HasError() {
+			return ps, fmt.Errorf("failed to configure provider: %v", diag)
+		}
+
+		ps.Meta = provider.TerraformProvider.Meta()
+
+		// This configures the Plugin Framework provider
+		schemaResponse := &terraformframework.SchemaResponse{}
+		provider.TerraformPluginFrameworkProvider.Schema(context.TODO(), terraformframework.SchemaRequest{}, schemaResponse)
```

Comment:
I'm not actually sure if this bit is necessary, it doesn't seem to solve the problem with generating controllers for Plugin Framework resources, but it also seems like something that might be needed in the end - it is needed for SDKv2 resources, so I assume TF Plugin Framework resources will need similar config down the line.